managed-ledger/src/main/java/org/apache/bookkeeper/mledger/util/RangeCache.java

```diff
@@ -175,6 +177,36 @@ public Value get(Key key) {
         return Pair.of(removedEntries, removedSize);
     }
 
+    /**
+    *
+    * @param minSize
+    * @return a pair containing the number of entries evicted and their total size
+    */
+   public Pair<Integer, Long> evictLEntriesBeforeTimestamp(long maxTimestamp) {
+       long removedSize = 0;
+       int removedEntries = 0;
+
+       while (true) {
+           Map.Entry<Key, Value> entry = entries.firstEntry();
+           if (entry == null || timestampExtractor.getTimestamp(entry.getValue()) > maxTimestamp) {
+               break;
+           }
+
+           entry = entries.pollFirstEntry();
+           if (entry == null) {
+               break;
+           }
+
+           Value value = entry.getValue();
+           ++removedEntries;
+           removedSize += weighter.getSize(value);
+           value.release();
+       }
+
+       size.addAndGet(-removedSize);
+       return Pair.of(removedEntries, removedSize);
```

Comment:
seems like only `removedSize` is used by the caller. So, we can just return `long` instead creating temporary Pair object for now. may be we can add back when we really need size and entries both?

Reply:
Good point. Fixed